modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/PartitionListener.java

```diff
@@ -485,58 +283,57 @@ private SingleRowResponse handleGetAndUpsertCommand(GetAndUpsertCommand cmd, lon
      * @return Result.
      */
     private boolean handleFinishTxCommand(FinishTxCommand cmd, long commandIndex) {
-        return storage.delegate().runConsistently(() -> {
-            UUID txId = cmd.txId();
-
-            boolean stateChanged = txManager.changeState(txId, TxState.PENDING, cmd.finish() ? TxState.COMMITED : TxState.ABORTED);
-
-            LockManager lockManager = txManager.lockManager();
-
-            // This code is technically incorrect and assumes that "stateChanged" is always true.
-            // This was done because transaction state is not persisted
-            // and thus FinishTxCommand couldn't be completed on recovery after node restart ("changeState" uses "replace").
-            if (/*txManager.state(txId) == TxState.COMMITED*/cmd.finish()) {
-                lockManager.locks(txId)
-                        .forEachRemaining(
-                                lock -> {
-                                    storage.commitWrite((ByteBuffer) lock.lockKey().key(), txId);
-                                }
-                        );
-            } else /*if (txManager.state(txId) == TxState.ABORTED)*/ {
-                lockManager.locks(txId)
-                        .forEachRemaining(
-                                lock -> {
-                                    storage.abortWrite((ByteBuffer) lock.lockKey().key());
-                                }
-                        );
-            }
+        UUID txId = cmd.getTxId();
+
+        boolean stateChanged = txManager.changeState(txId, TxState.PENDING, cmd.finish() ? TxState.COMMITED : TxState.ABORTED);
+
+        LockManager lockManager = txManager.lockManager();
+
+        // This code is technically incorrect and assumes that "stateChanged" is always true.
+        // This was done because transaction state is not persisted
+        // and thus FinishTxCommand couldn't be completed on recovery after node restart ("changeState" uses "replace").
+        if (/*txManager.state(txId) == TxState.COMMITED*/cmd.finish()) {
+            lockManager.locks(txId)
+                    .forEachRemaining(
+                            lock -> {
+                                storage.commitWrite((RowId) lock.lockKey().key(), new Timestamp(txId));
```

Comment:
Need to commit and abort only if lock is acquired for modification data.
Seems that lockManager.locks(txId) can return lock which is acquired on table and other entities.

Reply:
This code is invalid, until we won't fix a transactional commit.